src/ctapipe_io_nectarcam/__init__.py

```diff
@@ -92,6 +92,123 @@ class PixelStatus(IntFlag):
+    '''
+    Input:
+        camera geometry  
+            e.g. `source.subarray.tel[10].camera.geometry`
+    Returns:
+        pix_mid_module
+            list of pixels at the centre of each module (sorted)
+        
+    Requires the geometry to have pixel_x and pixel_y, and to have 7-pixel modules.
+    Neighbours gets lazy-loaded by the main ctapipe function in camera.geometry.
+    
+    Find central pixels of all the modules, as follows:
+
+    Finds the most distant PMT and which still has 6 neighbours, 
+      which must be the central pixel of a module.
+    Then elimates that module and repeats, until no modules left.
+
+    '''
+    
+    import numpy as np
+    
+    # Bug out right away if the number of pixels is not divisible by 7 (for 7-pixel modules)
+    if geometry.n_pixels%7 != 0:
+        raise ValueError("n_pixels is not divisible by 7, so not full 7-pixel modules. \n"+
+            "... Incompatible with expected NectarCAM geometry.")
+    
+    pix_mid_module = []
+    
+    pixel_dist = np.hypot(geometry.pix_x,geometry.pix_x).value
+
+    from copy import copy,deepcopy
+    # Make a dictionary by pixel index, so that pixels can be deleted later
+    pixel_dist_dict = dict(zip(range(geometry.n_pixels),pixel_dist))
+    
+    # geometry.neighbours gets lazy-loaded by the main ctapipe function in camera.geometry
+    pixel_nbrs_dict = dict(zip(range(geometry.n_pixels),deepcopy(geometry.neighbors)))
+
+    while len(pixel_dist_dict)>0:
+
+        # Find the farthest pixel which still has 6 neighbours
+        
+        # This holds the index within the dictionary
+        pixdist_order = np.argsort(list(pixel_dist_dict.values()))
+
+        for pix_idx in pixdist_order[::-1]: # Go backwards
+            pix_key = list(pixel_dist_dict.keys())[pix_idx]
+            if len(pixel_nbrs_dict[pix_key]) == 6:
+                break
+        pix_cent = pix_key
+        
+        #print("vvvvvvvvvvvvvvvvvvvvvvvvvvv",len(pixel_dist_dict))
+        #print("Drawer centre is :",pix_cent)
+        pix_mid_module.append(pix_cent)
+        
+        module = [pix_cent] + pixel_nbrs_dict[pix_key]
+        #print(f"Module: {module}")
+
+        # remove all the current module pixels from the neighbours list values
+        for pix_key in module:
+            #print(pix_key)
+            for nbr_key in pixel_nbrs_dict:
+                if pix_key in pixel_nbrs_dict[nbr_key]:
+                    #print(f"removing {pix_key} from {nbr_key}, {pixel_nbrs_dict[nbr_key]}")
+                    pixel_nbrs_dict[nbr_key].remove(pix_key)
+
+        # remove the current module pixels from the neighbours dictionary itself
+        for pix_key in module:
+            if pix_key in pixel_nbrs_dict:
+                pixel_nbrs_dict.pop(pix_key)
+        # remove the current module pixels from the distance dictionary
+        for pix_key in module:
+            if pix_key in pixel_dist_dict:
+                pixel_dist_dict.pop(pix_key)
```

Comment:
To avoid looping several times over `module`, this can be shortened by:
```
        for pix_key in module:
            # remove the current module pixels from the neighbours dictionary itself
            if pix_key in pixel_nbrs_dict:
                pixel_nbrs_dict.pop(pix_key)
            # remove the current module pixels from the distance dictionary
            if pix_key in pixel_dist_dict:
                pixel_dist_dict.pop(pix_key)
```